hearth/lib/hearth/middleware/send.rb

```diff
@@ -79,15 +67,31 @@ def call(input, context)
 
       private
 
+      def stub_response(input, context, output)
+        stub = @stubs.next(context.operation_name)
+        apply_stub(stub, input, context, output)
+        return unless context.response.body.respond_to?(:rewind)
+
+        context.response.body.rewind
+      end
+
+      def send_request(context, output)
+        @client.transmit(
+          request: context.request,
+          response: context.response,
+          logger: context.logger
+        )
+      rescue Hearth::NetworkingError => e
+        output.error = e
+      end
+
       def apply_stub(stub, input, context, output)
         case stub
         when Proc
           stub = stub.call(input, context)
           apply_stub(stub, input, context, output) if stub
-        when Exception
+        when Exception, ApiError
           output.error = stub
-        when Class
```

Comment:
We should update the docuemtnation in client_stubs to match whatever we decide here (just a note for future us).